### DISCUSSION TITLE
how to test whether an interface is defined？

### DISCUSSION THREAD

**Discussion by ljm0451** [google/googletest#3382] [Community Help]
Thanks advance
I'm a newer.
for example, I want to test whether my current libc provides the hello_world() interface.

I think the macro TEST in gtest can not do this. The reason is that if there is no hello_world() interface, gtest will compile error. I want to receive a PASS or not PASS, rather than a compile/link error.

what should I do?
-----------------------------------------------------------------------update 2021年4月25日13:21:10
I want to use gtest to test the posix interface, just like using "open posix test suite" . (For example, if OS/glibc does not support a certain interface, it should tell the user "not supported" or "not passed" instead of compiling/linking errors)

**Comment by XuhuaHuang** [comment]
Before jumping into macro `TEST`, have you tried with a `try-catch` block?
